modules/serverless-kn-containersource.adoc

```diff
@@ -0,0 +1,43 @@
+[id="serverless-kn-containersource_{context}"]
+= Creating and managing container sources by using the Knative CLI
+
+You can use the following `kn` commands to create and manage container sources:
+
+.Create a container source
+[source,terminal]
+----
+$ kn source container create <container_source_name> --image <image_uri> --sink <sink_uri>
```

Comment:
The `<sink_uri>` in this example isn't making it justice. The sink parameter is much more complex. And I think that complexity should be documented here.

>   -s, --sink string  Addressable sink for events. You can specify a broker, channel, Knative service or URI. Examples: '--sink broker:nest' for a broker 'nest', '--sink channel:pipe' for a channel 'pipe', '--sink ksvc:mysvc:mynamespace' for a Knative service 'mysvc' in another namespace 'mynamespace', '--sink https://event.receiver.uri' for an URI with an 'http://' or 'https://' schema, '--sink ksvc:receiver' or simply '--sink receiver' for a Knative service 'receiver' in the current namespace. If a prefix is not provided, it is considered as a Knative service in the current namespace. If referring to a Knative service in another namespace, 'ksvc:name:namespace' combination must be provided explicitly.

Reply:
This is _way_ outside the scope of container source docs since it applies for all `kn` commands that use the `--sink` flag.

We have the beginning of some reference docs here for this https://docs.openshift.com/container-platform/4.7/serverless/cli_tools/kn-flags-reference.html#specifying-sink-flag-kn_kn-flags-reference - please raise a separate Jira and include the info from your comment here about what needs to be added to it.

Like I said, this is way out of scope for this PR. It would be nice to link/xref out to the docs I mentioned for `--sink` here, and that's sort of the plan long term, but it's not something we can do until it's accepted to add xrefs to modules in OCP docs, which is something that's in progress.

Reply:
Unfortunately, not all `--sink` are the same. They should be to remove confusion. But until they aren't, we need to describe those parameters for each occurrence separately, which also means here.

Reply:
> Unfortunately, not all --sink are the same. They should be the same, to remove confusion about them. That would also allow to document them in one place. But they are not!

@cardil can you please elaborate a bit where they are not the same ? The client and the plugins are using a shared library for parsing the sink parameter, and if there are difference in the usage then it should be fixed asap. `--sink` has to have the same syntax everywhere (for configuring a KReference). If not either the option should be renamed or the option should be adapted to support the full syntax.

Reply:
`kn source kafka`, `kn trigger create`, `kn subscription create`, ... all have the same `--sink` option with the same syntax (either URI or prefixed for an Addressable)

Reply:
Good question @rhuss. I believe I saw somewhere, and it was different. I even used that to implement, the `--to` param in kn-event: 

https://github.com/knative-sandbox/kn-plugin-event/blob/0e1fe36db37682a10548808563a38c7e5ac3814c/cmd/kn-event/cmd/send.go#L40-L46

This field is being passed to https://pkg.go.dev/knative.dev/client@v0.24.0/pkg/util#ToTrackerReference method.

I don't remember what it was, unfortunately...